src/ripple/app/misc/SHAMapStoreImp.cpp

```diff
@@ -326,17 +326,14 @@ SHAMapStoreImp::run()
             lastRotated = validatedSeq;
             state_db_.setLastRotated (lastRotated);
         }
-        LedgerIndex canDelete = std::numeric_limits <LedgerIndex>::max();
-        if (setup_.advisoryDelete)
```

Comment:
This:
if (setup_.advisoryDelete)
    canDelete = state_db_.getCanDelete();

needs to move to the beginning of run(), and execute once after rippled startup, to initialize the value to what's been persisted.